contracts/modules/Experimental/TransferManager/LockupVolumeRestrictionTM.sol

```diff
@@ -229,164 +282,104 @@ contract LockupVolumeRestrictionTM is ITransferManager {
+                return unLockedAmount;
+            }
+            
+    }
 
     /**
      * @notice Parameter checking function for creating or editing a lockup.  This function will cause an exception if any of the parameters are bad.
-     * @param lockUpPeriodSeconds Total period of lockup (seconds)
-     * @param releaseFrequencySeconds How often to release a tranche of tokens (seconds)
-     * @param totalAmount Total amount of locked up tokens
+     * @param _userAddress Address whom lockup is being applied
+     * @param _lockupAmount Amount that needs to be locked
+     * @param _lockUpPeriodSeconds Total period of lockup (seconds)
+     * @param _releaseFrequencySeconds How often to release a tranche of tokens (seconds)
      */
-    function _checkLockUpParams(uint lockUpPeriodSeconds, uint releaseFrequencySeconds, uint totalAmount) internal view {
-        require(lockUpPeriodSeconds != 0, "lockUpPeriodSeconds cannot be zero");
-        require(releaseFrequencySeconds != 0, "releaseFrequencySeconds cannot be zero");
-        require(totalAmount != 0, "totalAmount cannot be zero");
+    function _checkLockUpParams(
+        address _userAddress,
+        uint256 _lockupAmount,
+        uint256 _lockUpPeriodSeconds,
+        uint256 _releaseFrequencySeconds
+    ) 
+        internal 
+        view 
+    {   
+        require(_userAddress != address(0), "Invalid address");
+        require(_lockUpPeriodSeconds != 0, "lockUpPeriodSeconds cannot be zero");
+        require(_releaseFrequencySeconds != 0, "releaseFrequencySeconds cannot be zero");
+        require(_lockupAmount != 0, "lockupAmount cannot be zero");
 
         // check that the total amount to be released isn't too granular
         require(
-            totalAmount % ISecurityToken(securityToken).granularity() == 0,
+            _lockupAmount % ISecurityToken(securityToken).granularity() == 0,
             "The total amount to be released is more granular than allowed by the token"
         );
 
         // check that releaseFrequencySeconds evenly divides lockUpPeriodSeconds
         require(
-            lockUpPeriodSeconds % releaseFrequencySeconds == 0,
+            _lockUpPeriodSeconds % _releaseFrequencySeconds == 0,
             "lockUpPeriodSeconds must be evenly divisible by releaseFrequencySeconds"
         );
 
-        // check that totalPeriods evenly divides totalAmount
-        uint totalPeriods = lockUpPeriodSeconds.div(releaseFrequencySeconds);
-        require(
-            totalAmount % totalPeriods == 0,
-            "The total amount being locked up must be evenly divisible by the number of total periods"
-        );
-
         // make sure the amount to be released per period is not too granular for the token
-        uint amountPerPeriod = totalAmount.div(totalPeriods);
+        uint256 totalPeriods = _lockUpPeriodSeconds.div(_releaseFrequencySeconds);
+        uint256 amountPerPeriod = _lockupAmount.div(totalPeriods);
```

Comment:
To fix the issue I described above, maybe add `require(amountPerPeriod*totalPeriods == totalAmount);`

Reply:
yes that I said above